pydantic/_internal/_known_annotated_metadata.py

```diff
@@ -312,6 +312,40 @@ def val_func(v: Any) -> Any:
                 return v
 
             schema = cs.no_info_after_validator_function(val_func, schema)
+        elif isinstance(annotation, at.Not):
+            if hasattr(annotation.func, '__qualname__'):
+                func_name = annotation.func.__qualname__
+
+                def val_func(v: Any) -> Any:
+                    if annotation.func(v):  # noqa: B023
+                        raise PydanticCustomError(
+                            'not_operation_failed',
+                            f'Not of {func_name} failed',  # type: ignore  # noqa: B023
+                        )
+
+                    return v
+
+                schema = cs.no_info_after_validator_function(val_func, schema)
```

Comment:
Other than the `not` in the `val_func`, this is basically identical to the `Predicate` logic - can we consolidate this logic?